### PR TITLE
removed memory balast and using GOMEMLIMIT. OTL-2488

### PR DESCRIPTION
**Pull request by samiura**
**Description:** removed all instances of memory_ballast and using GOMEMLIMIT

**Link to Splunk idea:** [OTL-2488](https://splunk.atlassian.net/browse/OTL-2488)

**Testing:** added unit tests

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: helm-charts/splunk-otel-collector/templates/daemonset.yaml [hunk 1]
```diff
@@ -269,6 +269,8 @@ spec:
           {{- include "splunk-otel-collector.securityContext" (dict "isWindows" .Values.isWindows "securityContext" $agent.securityContext "setRunAsUser" true) | nindent 10 }}
         {{- end }}
         env:
+          - name: GOMEMLIMIT
+            value: {{ include "splunk-otel-collector.convertMemToMib" $agent.resources.limits.memory | int64 }}
           - name: SPLUNK_MEMORY_TOTAL_MIB
```
Comment (pjanotti) [inline helm-charts/splunk-otel-collector/templates/daemonset.yaml:274]: Shouldn't `SPLUNK_MEMORY_TOTAL_MIB` be removed? If not how it interacts with `GOMEMLIMIT`?

**Review by pjanotti — APPROVED** [review]
LGTM, but, I have one Q.

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 2]
```diff
@@ -17,8 +17,6 @@ extensions:
     timeout: 0
   {{- end }}
 
-  memory_ballast:
-    size_mib: ${SPLUNK_BALLAST_SIZE_MIB}
```
Comment (jvoravong) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:20]:
```suggestion
```

File: helm-charts/splunk-otel-collector/templates/daemonset.yaml [hunk 3]
```diff
@@ -269,6 +269,8 @@ spec:
           {{- include "splunk-otel-collector.securityContext" (dict "isWindows" .Values.isWindows "securityContext" $agent.securityContext "setRunAsUser" true) | nindent 10 }}
         {{- end }}
         env:
+          - name: GOMEMLIMIT
+            value: {{ include "splunk-otel-collector.convertMemToMib" $agent.resources.limits.memory | int64 }}
```
Comment (jvoravong) [inline helm-charts/splunk-otel-collector/templates/daemonset.yaml:273]: The test might be failing here due to a quotation or some other issue, likely also affects deployment-gateway.yaml.
```suggestion
            value: "{{ include "splunk-otel-collector.convertMemToMib" $agent.resources.limits.memory | int64 }}"
```

File: helm-charts/splunk-otel-collector/templates/daemonset.yaml (hunk 1 above)
Comment (samiura) [reply helm-charts/splunk-otel-collector/templates/daemonset.yaml:274]: As far as I understand it has no effect from the tests on the internal realms. this is a good point, may be we can investigate the idea of removing it.

File: .chloggen/.OTL2488.yaml.un~
Comment (dmitryax) [inline .chloggen/.OTL2488.yaml.un~:1]: Please remove this file :)

File: helm-charts/splunk-otel-collector/templates/daemonset.yaml (hunk 1 above)
Comment (dmitryax) [reply helm-charts/splunk-otel-collector/templates/daemonset.yaml:274]: If we can set it dynamically in https://github.com/signalfx/splunk-otel-collector/blob/91f039d015bdd5792e172ea635bc9b1dfd07a101/internal/settings/settings.go from `SPLUNK_MEMORY_TOTAL_MIB`, that would be great. So we don't need to provide two separate env vars. I remember @atoulme mentioned that he tried that.

**Pull request closed without merging**